lib/matplotlib/axes/_axes.py

```diff
@@ -2779,8 +2779,9 @@ def stem(self, *args, linefmt=None, markerfmt=None, basefmt=None, bottom=0,
 
         # New behaviour in 3.1 is to use a LineCollection for the stemlines
         if use_line_collection:
-            stemlines = []
             stemlines = [((xi, bottom), (xi, yi)) for xi, yi in zip(x, y)]
+            if linestyle is None:
+                linestyle = 'solid'
```

Comment:
Should this maybe rather evaluate the `lines.linestyle` rc parameter?

Reply:
Yes. Thanks!